Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,143 @@
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public override void Update(BaseData data)
+        {
+            var rate = data.Price;
+            var dataType = data.GetType();
+
+            decimal volume;
+            decimal open;
+            decimal high;
+            decimal low;
+            decimal close;
+
+            if (dataType == typeof(TradeBar))
+            {
+                volume = ((TradeBar)data).Volume;
+                open = ((TradeBar)data).Open;
+                high = ((TradeBar)data).High;
+                low = ((TradeBar)data).Low;
+                close = rate;
+            }
+            else if (dataType == typeof(Tick))
+            {
+                // Only include actual trade information
+                if (((Tick)data).TickType != TickType.Trade)
+                {
+                    return;
+                }
+
+                volume = ((Tick)data).Quantity;
+                open = rate;
+                high = rate;
+                low = rate;
+                close = rate;
+            }
+            else
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
+
+            if (_firstTick)
+            {
+                _firstTick = false;
+                _currentBar = new VolumeRenkoBar(data.Symbol, data.Time, data.EndTime, _barSize, open, high, low, close, volume);
+            }
+            else
+            {
+                _volumeLeftOver = _currentBar.Update(data.EndTime, high, low, close, volume);
+                while (_volumeLeftOver >= 0)
+                {
+                    OnDataConsolidated(_currentBar);
+                    _currentBar = new VolumeRenkoBar(data.Symbol, data.EndTime, data.EndTime, _barSize, close, high, low, close, 0);
+                    _volumeLeftOver = _currentBar.Update(data.EndTime, high, low, close, _volumeLeftOver);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Scans this consolidator to see if it should emit a bar due to time passing
+        /// </summary>
+        /// <param name="currentLocalTime">The current time in the local time zone (same as <see cref="BaseData.Time"/>)</param>
+        public override void Scan(DateTime currentLocalTime)
+        {
+        }
+
+        /// <summary>
+        /// Event invocator for the DataConsolidated event. This should be invoked
+        /// by derived classes when they have consolidated a new piece of data.
+        /// </summary>
+        /// <param name="consolidated">The newly consolidated data</param>
+        protected void OnDataConsolidated(VolumeRenkoBar consolidated)
+        {
+            DataConsolidated?.Invoke(this, consolidated);
+            Consolidated = consolidated;
+        }
```

Comment:
Instead of having the method can directly call base `OnDataConsolidated` ?

Reply:
I'll inherit the base virtual method directly

Reply:
The input type is different?